File: src/dataset/cpp/collatz-conjecture_exercism-toemm/claude/ClaudeSonnet4_collatz_conjecture_v2.cpp
#include "collatz_conjecture.h"

#include <stdexcept>

namespace collatz_conjecture {

int steps(int i) {
    if (i <= 0) throw std::domain_error("i not positive and above 0");
    
    int ctr = 0;
    
    while (i > 1) {
        if (i & 1) {
            i = i * 3 + 1;
        } else {
            i >>= 1;
        }
        ctr++;
    }
    
    return ctr;
}

}  // namespace collatz_conjecture

File: src/dataset/typescript/collatz-conjecture_exercism-samajammin/claude/ClaudeSonnet4_collatz-conjecture_v2.ts
class CollatzConjecture {
  static steps(int: number): number {
    if (int <= 0) {
      throw new Error('Only positive numbers are allowed');
    }
    let steps: number = 0;
    while (int !== 1) {
      if (int & 1) {
        int = int * 3 + 1;
      } else {
        int >>>= 1;
      }
      steps++;
    }
    return steps;
  }
}

export default CollatzConjecture;